test/sequential/test-net-connect-local-error.js

```diff
@@ -32,10 +32,10 @@ function onError(err, options) {
   );
 }
 
-const clientIPv4 = net.connect(optionsIPv4);
-clientIPv4.on('error', common.mustCall((err) => onError(err, optionsIPv4)));
-
 if (!common.hasIPv6) {
+  const clientIPv4 = net.connect(optionsIPv4);
+  clientIPv4.on('error', common.mustCall((err) => onError(err, optionsIPv4)));
+
```

Comment:
We shouldn't do that, IPv4 should work everywhere, not just where IPv6 is not available.

Reply:
IPv4 fails with `ERR_ASSERTION`.

Reply:
Well we need to fix that. If this PR breaks IPv4, it would almost certainly disrupt the ecosystem in a bad way.

Reply:
Sure. I want to see if it passes the IPv6 part, though. But I also don't know why the `connect` fails on an IPv6-enabled system.